File: Pset1/mario.c
/*
 * mario.c
 * Aamel Unia
 * aameluni@buffalo.edu
 * Simple program that forms a half-pyramid in which hashes represent blocks, similar to the half-pyramid found toward the end 
   of World 1-1 in Nintendo’s Super Mario Brothers 
 * The half-pyramid’s height which is entered by the user must be a non-negative integer no greater than 23.
 */


#include <cs50.h>
#include <stdio.h>

int main (void){


int n;
do{
    printf("Height: ");
    n=GetInt();
}
while (n<0||n>23);

int spaces = n - 1;
int temp = spaces;
    
for (int i=0; i<n; i++){
    
    for(; spaces!=0; spaces--){
        printf(" ");
    }

    for (int j=i+2;j>0;j--){
    printf("#");
    }
    
    printf("\n");
    
    spaces = temp;
    temp --;
    spaces--;
}

    
    
  return 0;  
}

File: Pset6/mario.py
#   mario.py
#   Aamel Unia
#   aameluni@buffalo.edu
#   Simple program that prints out a double half-pyramid of a specified height in which hashes represent blocks, similar to the 
#   double half-pyramid found toward the end of World 1-1 in Nintendo’s Super Mario Brothers.
#   The half-pyramid’s height which is entered by the user must be a non-negative integer no greater than 23.
 
def main():
    
    while True:
        print("Height: ",end="")
        height = int(input())
        if height>0 and height<=23:
            break
    
    spaces = height-1
    
    for i in range(height):
        for x in range(spaces,0,-1):
           print(" ",end="")
        for z in range(i+1,0,-1):
            print("#",end="")
        print("  ",end="")
        for k in range(i+1,0,-1):
            print("#",end="")
        for y in range(spaces):
            print(" ",end="")
        print()
        spaces -= 1
        


if __name__ == "__main__":
    main()
    